src/Grpc.AspNetCore.Server/IGrpcServiceActivator.cs

```diff
@@ -27,13 +29,14 @@ public interface IGrpcServiceActivator<TGrpcService> where TGrpcService : class
         /// <summary>
         /// Creates a service.
         /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
         /// <returns>The created service.</returns>
-        TGrpcService Create();
+        GrpcActivatorHandle<TGrpcService> Create(IServiceProvider serviceProvider);
 
         /// <summary>
         /// Releases the specified service.
         /// </summary>
         /// <param name="service">The service to release.</param>
-        void Release(TGrpcService service);
+        void Release(in GrpcActivatorHandle<TGrpcService> service);
```

Comment:
Would `ref` be better than `in` so we can set the handle back to `default`?

I think it also makes more sense semantically, since `in` generally means that the Handle is unmodified, but really the handle isn't usable after you call this.

Reply:
Fair, but we're going to make this internal so API shape isn't SUPER important right now.